kie-wb-common-react/boxed-expression-editor/src/components/ExpressionContainer/ExpressionContainer.tsx

```diff
@@ -44,9 +38,16 @@ export const ExpressionContainer: ({ selectedExpression }: ExpressionContainerPr
   const [logicTypeSelected, setLogicTypeSelected] = useState(
     !_.isEmpty(props.selectedExpression.logicType) || props.selectedExpression.logicType === LogicType.Undefined
   );
-  const [actionDropdownOpen, setActionDropDownOpen] = useState(false);
   const [selectedExpression, setSelectedExpression] = useState(props.selectedExpression);
 
+  const [
+    contextMenuRef,
+    contextMenuXPos,
+    contextMenuYPos,
+    contextMenuIsVisible,
+    setContextMenuVisibility,
+  ] = useContextMenuHandler();
+
```

Comment:
I believe it is be better to set event listeners at each re-render, because as you are stating, undo/redo will be supported for sure. Thanks for thinking at whole picture. Really helpful.

Reply:
So if there's the undo/redo possibility, then I suppose you have to check for props change with an `useEffect` and then update `logicTypeSelected` and `selectedExpression` accordingly. Otherwise the user will not see changes.

Reply:
There is a dedicated task for managing undo/redo, that will impact the whole editor. I will manage it accordingly after designing that solution.